src/algorithms/reco/ElectronReconstruction.cc

```diff
@@ -0,0 +1,91 @@
+
+// Event Model related classes
+#include "edm4hep/MCParticleCollection.h"
+#include "edm4eic/MCRecoParticleAssociationCollection.h"
+#include "edm4eic/ReconstructedParticleCollection.h"
+
+namespace eicrecon {
+
+  void ElectronReconstruction::init(std::shared_ptr<spdlog::logger> logger) {
+    m_log = logger;
+  }
+
+  std::vector<edm4eic::ReconstructedParticle*> ElectronReconstruction::execute(
+    std::vector<const edm4hep::MCParticle *> mcparts,
+    std::vector<const edm4eic::ReconstructedParticle *> rcparts,
+    std::vector<const edm4eic::MCRecoParticleAssociation *> rcassoc,
+    std::vector<std::vector<const edm4eic::MCRecoClusterParticleAssociation*>> &in_clu_assoc
+    ) {
+
+        // Step 1. Loop through MCParticle - cluster associations
+        // Step 2. Get Reco particle for the Mc Particle matched to cluster
+        // Step 3. Apply E/p cut using Reco cluster Energy and Reco Particle momentum
+
+        // Some obvious improvements:
+        // - E/p cut from real study optimized for electron finding and hadron rejection
+        // - use of any HCAL info?
+        // - check for duplicates?
+
+        m_log->debug("ElectronReconstruction::execute");
+
+        // output container
+        std::vector<edm4eic::ReconstructedParticle*> electrons_edm;
+
+        for ( auto col : in_clu_assoc ){ // loop on cluster association collections
+          for ( auto clu_assoc : col ){ // loop on MCRecoClusterParticleAssociation in this particular collection
+            auto sim = clu_assoc->getSim(); // McParticle
+            auto clu = clu_assoc->getRec(); // RecoCluster
+
+            m_log->debug( "SimId={}, CluId={}", clu_assoc->getSimID(), clu_assoc->getRecID() );
+            m_log->debug( "MCParticle: Energy={}, p={}, E/p = {} for PDG: {}", clu.getEnergy(), edm4eic::magnitude(sim.getMomentum()), clu.getEnergy() / edm4eic::magnitude(sim.getMomentum()), sim.getPDG() );
+
+
+            // Find the Reconstructed particle associated to the MC Particle that is matched with this reco cluster
+            // i.e. take (MC Particle <-> RC Cluster) + ( MC Particle <-> RC Particle ) = ( RC Particle <-> RC Cluster )
+            auto reco_part_assoc = rcassoc.begin();
+            bool found_reco_part = false;
+            for (; reco_part_assoc != rcassoc.end(); ++reco_part_assoc) {
+              if ((*reco_part_assoc)->getSimID() == (unsigned) clu_assoc->getSimID()) {
+                found_reco_part = true;
+                break;
+              }
+            }
+
+            // if we found a reco particle then test for electron compatibility
+            if ( found_reco_part ){
+              auto reco_part = (*reco_part_assoc)->getRec();
+              double EoverP = clu.getEnergy() / edm4eic::magnitude(reco_part.getMomentum());
+              m_log->info( "ReconstructedParticle: Energy={}, p={}, E/p = {} for PDG (from truth): {}", clu.getEnergy(), edm4eic::magnitude(reco_part.getMomentum()), EoverP, sim.getPDG() );
```

Comment:
Logs from inside loops should be even below debug.
```suggestion
              m_log->trace( "ReconstructedParticle: Energy={}, p={}, E/p = {} for PDG (from truth): {}", clu.getEnergy(), edm4eic::magnitude(reco_part.getMomentum()), EoverP, sim.getPDG() );
```